nautobot_circuit_maintenance/handle_notifications/sources.py

```diff
@@ -193,9 +215,53 @@ def receive_notifications(
         self.close_session()
         return received_notifications
 
+    def validate_providers(self, logger: Job, notification_source: NotificationSource, since_txt: str) -> bool:
+        """Method to validate that the NotificartionSource has attached Providers.
+
+        Args:
+            logger (Job): Job to use its logger
+            notification_source (NotificationSource): Notification Source to validate providers
+            since_txt (str): Date string to be used to log
+
+        Returns:
+            bool: True if there are relevant providers attached or False otherwise
+        """
+        providers_with_email = []
+        providers_without_email = []
+        if not notification_source.providers.all():
+            logger.log_warning(
+                message="Skipping this email account no providers were defined.",
+            )
+            return False
+
+        for provider in notification_source.providers.all():
+            for custom_field, value in provider.get_custom_fields().items():
+                if custom_field.name == "emails_circuit_maintenances" and value:
+                    self._emails_to_fetch.extend(value.split(","))
+                    providers_with_email.append(provider.name)
+                else:
+                    providers_without_email.append(provider.name)
+
+        if providers_without_email:
+            logger.log_warning(
```

Comment:
Possibly change this one to `log_info`? I suspect it'll be pretty normal for us to have some providers without email.

Reply:
I defined as warning because if those providers are "attached" to this NotificationSource, it doesn't make any sense because without a `source_email` none of the emails fetched by this IMAP source will be related to this provider, so it would like to signal it as warning to catch the attention from the operator.
Note: the providers taken into account are the ones related to this NotificationSource, that has been instantiated as IMAP. Other source types, for instance a rest API, would not use the same `validate_providers` logic, will have a custom one that won't check for  `source_emails`